### DISCUSSION TITLE
use of several github actions to build for different os versions

### DISCUSSION THREAD

**Discussion by ShortyBoa** [NightscoutFoundation/xDrip#3696] [General]
Hey folks,

within the last time I had the chance to look through the source code and was trying to figure out, wear os 4 and later are not yet fully compatible.

I'm using a Samsung 21 5G FE currently on Android13 Tiramisu and got a Samsung Galaxy Watch 6 LTE. Before the android os upgrade, I used the wear as a standalone collector with my Dexcom G6 transmitter. This setup was perfect, giving me the possibility to leave the phone at home and still being available on my watch, while the watch also retrieved all values. Since the upgrade to tiramisu, this setup broke.

Since xDrip is used by a lot of people all over the globe, where not everyone has access to newer technologies, all the ones with newer devices has to take a downer on functionality.

Since the github action is only a build step, why don't we improve the possiblites a interchaging android manifests and build different os releases?

Best regards
Shorty

**Comment by Navid200** [comment]
All of this is only my personal view:  

We already have a hard time supporting a single release.
We don't have the bandwidth to support more.
We rely on the community to use the Nightly releases so that we can see what's wrong and needs attention.  Yet, every time we release a new stable release, we have to worry if there will be new complaints because the majority only use the stable release.
That will only get worse if we split the community and have half use one release and the other half, the other release.

WearOS is more like a luxury -  I cannot emphasize enough that this is only my personal view.
It is not like you cannot see your glucose any longer.  You only cannot see it on your watch.  

WearOS is not supported as well as it used to.
There is so much to do to keep xDrip going.   A few developers who are actively working in this repository dedicate their limited time to what they decide really needs attention.
Items that are critical get attention.  For example, xDrip Sync broke a few months ago.  That directly affected parents following their children.  That was extremely critical.  Attention was given to that to fix it.  And it is still being verified.  I cannot imagine if we had to verify it in two different optional releases.

**Reply by ShortyBoa** [reply]
Okay, with that in mind, the major issues are the lack of integration tests, when a build has been created?

Would it help, if a devops engineer hooks up to this point, so the developing community can identify incidents earlier?

I don't mind to building the chain. I just need time to get into gradle and the android and java SDKs. We probably have to stick our heads together for that I'll get an overview of actual needs.

**Reply by Navid200** [reply]
I cannot speak for JamOrHam.

I am happy to help if I can.
